include/LightGBM/c_api.h

```diff
@@ -10,55 +18,45 @@
 #include <cstdint>
 #include <cstring>
 
-/*!
-* To avoid type conversion on large data, most of our expose interface support both for float_32 and float_64.
-* Except following:
-* 1. gradients and hessians.
-* 2. Get current score for training data and validation
-* The reason is because they are called frequently, the type-conversion on them maybe time cost.
-*/
 
-typedef void* DatasetHandle;
-typedef void* BoosterHandle;
+typedef void* DatasetHandle;  /*!< \brief Handle of dataset. */
+typedef void* BoosterHandle;  /*!< \brief Handle of booster. */
 
-#define C_API_DTYPE_FLOAT32 (0)
-#define C_API_DTYPE_FLOAT64 (1)
-#define C_API_DTYPE_INT32   (2)
-#define C_API_DTYPE_INT64   (3)
-#define C_API_DTYPE_INT8    (4)
+#define C_API_DTYPE_FLOAT32 (0)  /*!< \brief float32 (single precision float). */
+#define C_API_DTYPE_FLOAT64 (1)  /*!< \brief float64 (double precision float). */
+#define C_API_DTYPE_INT32   (2)  /*!< \brief int32. */
+#define C_API_DTYPE_INT64   (3)  /*!< \brief int64. */
+#define C_API_DTYPE_INT8    (4)  /*!< \brief int8. */
 
-#define C_API_PREDICT_NORMAL     (0)
-#define C_API_PREDICT_RAW_SCORE  (1)
-#define C_API_PREDICT_LEAF_INDEX (2)
-#define C_API_PREDICT_CONTRIB    (3)
+#define C_API_PREDICT_NORMAL     (0)  /*!< \brief Normal prediction, with transform (if needed). */
+#define C_API_PREDICT_RAW_SCORE  (1)  /*!< \brief Predict raw score. */
+#define C_API_PREDICT_LEAF_INDEX (2)  /*!< \brief Predict leaf index. */
+#define C_API_PREDICT_CONTRIB    (3)  /*!< \brief Predict feature contributions (SHAP values). */
 
 /*!
- * \fn LGBM_GetLastError
- * \headerfile <LightGBM/export.h>
  * \brief Get string message of the last error.
- * \return error information
-*/
+ * \return Error information
+ */
 LIGHTGBM_C_EXPORT const char* LGBM_GetLastError();
 
 // --- start Dataset interface
 
 /*!
- * \fn LGBM_DatasetCreateFromFile
  * \brief Load dataset from file (like LightGBM CLI version does).
  * \param filename The name of the file
  * \param parameters Additional parameters
  * \param reference Used to align bin mapper with other dataset, nullptr means isn't used
  * \param[out] out A loaded dataset
  * \return 0 when succeed, -1 when failure happens
-*/
+ */
 LIGHTGBM_C_EXPORT int LGBM_DatasetCreateFromFile(const char* filename,
                                                  const char* parameters,
                                                  const DatasetHandle reference,
                                                  DatasetHandle* out);
 
 /*!
- * \fn LGBM_DatasetCreateFromSampledColumn
  * \brief Create an empty dataset by sampling data.
```

Comment:
Is it really empty?

Reply:
it will allocate the spaces, so no really empty. But the data are push afterward.

Reply:
Thanks, got it! Maybe then change it to `Allocate the space for dataset by sampling data.`?

Reply:
sure, BTW, it will also bucket the feature bins (construct bin_mapper) according to sample data.

Reply:
Updated. And one next function description too.